core/src/apps/ripple/serialize.py

```diff
@@ -12,30 +12,46 @@
 
 from . import helpers
 
+if False:
+    from trezor.utils import Writer
+
+
+class RippleField:
+    def __init__(self, type: int, key: int) -> None:
+        self.type: int = type
+        self.key: int = key
+
+
 FIELD_TYPE_INT16 = 1
 FIELD_TYPE_INT32 = 2
 FIELD_TYPE_AMOUNT = 6
 FIELD_TYPE_VL = 7
 FIELD_TYPE_ACCOUNT = 8
 
-FIELDS_MAP = {
-    "account": {"type": FIELD_TYPE_ACCOUNT, "key": 1},
-    "amount": {"type": FIELD_TYPE_AMOUNT, "key": 1},
-    "destination": {"type": FIELD_TYPE_ACCOUNT, "key": 3},
-    "fee": {"type": FIELD_TYPE_AMOUNT, "key": 8},
-    "sequence": {"type": FIELD_TYPE_INT32, "key": 4},
-    "type": {"type": FIELD_TYPE_INT16, "key": 2},
-    "signingPubKey": {"type": FIELD_TYPE_VL, "key": 3},
-    "flags": {"type": FIELD_TYPE_INT32, "key": 2},
-    "txnSignature": {"type": FIELD_TYPE_VL, "key": 4},
-    "lastLedgerSequence": {"type": FIELD_TYPE_INT32, "key": 27},
-    "destinationTag": {"type": FIELD_TYPE_INT32, "key": 14},
+FIELDS_MAP: dict[str, RippleField] = {
+    "account": RippleField(type=FIELD_TYPE_ACCOUNT, key=1),
+    "amount": RippleField(type=FIELD_TYPE_AMOUNT, key=1),
+    "destination": RippleField(type=FIELD_TYPE_ACCOUNT, key=3),
+    "fee": RippleField(type=FIELD_TYPE_AMOUNT, key=8),
+    "sequence": RippleField(type=FIELD_TYPE_INT32, key=4),
+    "type": RippleField(type=FIELD_TYPE_INT16, key=2),
+    "signingPubKey": RippleField(type=FIELD_TYPE_VL, key=3),
+    "flags": RippleField(type=FIELD_TYPE_INT32, key=2),
+    "txnSignature": RippleField(type=FIELD_TYPE_VL, key=4),
+    "lastLedgerSequence": RippleField(type=FIELD_TYPE_INT32, key=27),
+    "destinationTag": RippleField(type=FIELD_TYPE_INT32, key=14),
 }
 
 TRANSACTION_TYPES = {"Payment": 0}
 
 
-def serialize(msg: RippleSignTx, source_address: str, pubkey=None, signature=None):
+def serialize(
+    msg: RippleSignTx,
+    source_address: str,
+    pubkey: bytes | None = None,
```

Comment:
afaict `pubkey` is specified in all usages so please make it required

Reply:
Done in 329a038 - it just needed a small unittest modification to pass `None` there explicitly